src/tt_publisher.erl

```diff
@@ -0,0 +1,13 @@
+-module(tt_publisher).
+
+-export([publish/1]).
+
+-record(talk, {title :: string(), start_time :: calendar:datetime(), end_time :: calendar:datetime(), location :: string()}).
+
+-spec(publish(#talk{}) -> no_return()).
+publish(Talks) ->
```

Comment:
I dont think that this method needs TimeInterval argument that you suggested in. All functionality that has to do with time intervals between tweets can be encapsulated inside tt_scheduler module and this method shouldnt need any information about that.